tape_measure_tool.md

```diff
@@ -0,0 +1,90 @@
+# Tape Measure Tool
+
+This document outlines the design for the tape measure tool targeted at
+Ignition Dome and onwards.  The reasoning for the targeting of Dome is largely
+due to the snapping behavior which needs bounding boxes.  If this measurement
+tool is needed for Citadel, I can try to make a prototype version that doesn't
+have any of the more complex snapping capabilities and target it towards
+Citadel.
+
+## Purpose
+
+The tape measure tool, as its name implies, is a tool that aids the user in
+measurements of distance within a 3D scene. Currently, there is no method
+of extracting any sort of measurement data.  This sort of information is data
+that is quite useful to a user working within Ignition Gazebo, whether it be
+making various calculations or designing/creating a world and needing exact
+precision.  It can also just be simply because it is easy to lose the scale of
+the world and the objects within it when working within a simulator. This
+functionality will also be quite convenient with the eventual addition of
+entity scaling.  On a tangent, the dimensional data of an entity, which I
+assume will be a reason for which this tool will be used the most, could also
+likely be placed within the Component Inspector plugin so the user isn't left
+clicking all of the bounds of an entity and having to manually keep track of
+the entity's dimensions themselves.  There is an issue out for similar type
+improvements [here](https://github.com/ignitionrobotics/ign-gazebo/issues/158)
+where I expect to eventually add this information to the Component Inspector.
+
+## Functionality
+
+The tape measure tool is likely to be very simplistic in nature, but has
+potential to have certain nuances that this design doc is intended to flesh
+out.
+
+I expect the user experience to be something like the following:
+ 1. The user adds the Tape Measure Tool plugin via the plugin menu.
+ 2. At this point, I suppose there could be a button one could click
+    that is something like "New measurement" at which point would enable
+    the tape measure tool.  There could also be a hotkey to start a new
+    measurement so the user can quickly take measurements without having
+    to constantly click the "New Measurement" button.
+ 3. The mouse could then change into a cross-shaped icon and display a
+    transparent point which gives a preview of the point to be measured,
+    this point will become opaque once the user clicks.
+ 4. We have two options at this point, neither of which would differ
+    in behavior, we could either:
+    * Have a click system where the user clicks point A and then clicks
+      point B.
+    * Have a drag system where the user clicks on a particular point, holding
+      down the mouse and dragging the cursor to the desired end point.
```

Comment:
The click system is better, because it allows the user to orbit the world in between points